snuba/cli/offline_replacer.py

```diff
@@ -0,0 +1,119 @@
+import logging
+import time
+from datetime import datetime
+from typing import Optional
+
+import click
+import simplejson as json
+from arroyo import Message
+from arroyo.backends.kafka import KafkaPayload
+from arroyo.types import Partition, Topic
+
+from snuba import environment
+from snuba.datasets.storages import StorageKey
+from snuba.datasets.storages.factory import get_writable_storage
+from snuba.environment import setup_logging, setup_sentry
+from snuba.replacer import ReplacerWorker
+from snuba.utils.metrics.wrapper import MetricsWrapper
+
+
+@click.command()
+@click.option(
+    "--storage",
+    "storage_name",
+    type=click.Choice(["errors", "errors_v2"]),
+    help="The storage to consume/run replacements for (currently only events supported)",
+    required=True,
+)
+@click.option(
+    "--file",
+    "file_name",
+    help="The file name containing the replacement to be applied",
+    required=True,
+)
+@click.option(
+    "--project",
+    "project_id",
+    type=int,
+    help="The project to execute",
+    required=True,
+)
+@click.option("--dry-run", is_flag=True)
+@click.option(
+    "--start-from",
+    "start_from",
+    type=int,
+    help="Offset in lines from the beginning of the file",
+    default=0,
+)
+@click.option(
+    "--delay",
+    "delay",
+    type=int,
+    help="Waiting time between two replacements in seconds",
+    default=0,
+)
+@click.option("--log-level", help="Logging level to use.")
+def offline_replacer(
+    *,
+    storage_name: str,
+    file_name: str,
+    project_id: int,
+    dry_run: bool,
+    start_from: int,
+    delay: int,
+    log_level: Optional[str] = None,
+) -> None:
+    """
+    Executes a list of replacements taken from a file instead of taking
+    them from Kafka.
+    """
+    logger = logging.getLogger(__name__)
+    setup_logging(log_level)
+    setup_sentry()
+
+    metrics = MetricsWrapper(environment.metrics, "errors.replacer")
```

Comment:
I am not sure whether metrics would be reported when the script is run. But if they are going to be reported then maybe have a different prefix for metrics reported from script vs what is run by the consumer